optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverManager.java

```diff
@@ -0,0 +1,203 @@
+     * The planning problem is solved on a solver {@link Thread}, as soon as one is available.
+     * <p>
+     * When the solver terminates, the {@code finalBestSolutionConsumer} is called once with the final best solution,
+     * on a consumer {@link Thread}, as soon as one is available.
+     * To listen to intermediate best solutions too, use {@link #solveObserving(Object, Function, Consumer)} instead.
+     * <p>
+     * Defaults to logging exceptions as an error.
+     * <p>
+     * To stop a solver job before it naturally terminates, call {@link #terminateEarly(Object)}.
+     * @param problemId never null, a ID for each planning problem. This must be unique.
+     * Use this problemId to {@link #terminateEarly(Object) terminate} the solver early,
+     * {@link #getSolverStatus(Object) to get the status} or if the problem changes while solving.
+     * @param problemFinder never null, a function that returns a {@link PlanningSolution}, usually with uninitialized planning variables
+     * @param finalBestSolutionConsumer sometimes null, called only once, at the end, on a consumer thread
+     * @return never null
+     */
+    default SolverJob<Solution_, ProblemId_> solveBatch(ProblemId_ problemId,
+            Function<ProblemId_, Solution_> problemFinder, Consumer<Solution_> finalBestSolutionConsumer) {
+        return solveBatch(problemId, problemFinder, finalBestSolutionConsumer, null);
+    }
+
+    /**
+     * As defined by {@link #solveBatch(Object, Function, Consumer)}.
+     * @param problemId never null, a ID for each planning problem. This must be unique.
+     * Use this problemId to {@link #terminateEarly(Object) terminate} the solver early,
+     * {@link #getSolverStatus(Object) to get the status} or if the problem changes while solving.
+     * @param problemFinder never null, function that returns a {@link PlanningSolution}, usually with uninitialized planning variables
+     * @param finalBestSolutionConsumer sometimes null, called only once, at the end, on a consumer thread
+     * @param exceptionHandler sometimes null, called if an exception or error occurs.
+     * If null it defaults to logging the exception as an error.
+     * @return never null
+     */
+    SolverJob<Solution_, ProblemId_> solveBatch(ProblemId_ problemId,
+            Function<ProblemId_, Solution_> problemFinder, Consumer<Solution_> finalBestSolutionConsumer,
+            BiConsumer<ProblemId_, Throwable> exceptionHandler);
+
+    /**
+     * Submits a planning problem to solve and returns immediately.
+     * The planning problem is solved on a solver {@link Thread}, as soon as one is available.
+     * <p>
+     * When the solver finds a new best solution, the {@code bestSolutionConsumer} is called every time,
+     * on a consumer {@link Thread}, as soon as one is available (taking into account any throttling waiting time),
+     * unless a newer best solution is already available by then (in which case skip ahead discards it).
```

Comment:
"throttling time" and "skip ahead" are terms I have not seen described in the Javadocs.